internal/vendor/vendor.go

```diff
@@ -55,6 +57,14 @@ func New(
 	tokenVendor TokenVendor,
 ) PipelineTokenVendor {
 	return func(ctx context.Context, claims jwt.BuildkiteClaims, requestedRepoURL string) (*PipelineRepositoryToken, error) {
+		// Check for the existence of organization profiles in the current context
+		_, ok := ctx.Value("profileConfig").(github.ProfileConfig)
+		if !ok {
+			log.Info().Msg("No organization profile found. Continuing.")
+		}
+
+		// TODO: Check if the requested profile exists in the organization profile
```

Comment:
I could be misunderstanding something as I read this, but I'm pretty sure the `context.Context` instance used here is never updated, and will not receive either initial or refreshed data.